core/src/main/java/org/kohsuke/stapler/MetaClass.java

```diff
@@ -182,7 +182,7 @@ public String toString() {
             });
         }
 
-        FunctionList getMethods = node.methods.prefix("get");
+        FunctionList getMethods = node.methods.prefix("get").filter(m -> !m.getSignature().equals("method getClass"));
```

Comment:
`method java.lang.Object getClass` actually.

So it looks like this deserves a test after all.